### PR TITLE
Unit testing all the things

### PR DESCRIPTION
**Pull request by janssejl**

### CODE REVIEW

File: src/graph/auth/auth.spec.ts [hunk 1]
```diff
@@ -0,0 +1,40 @@
...
+import "reflect-metadata";
+import { Types } from "../../ioc/types";
+import { injectable } from "inversify";
+
+@injectable()
+export class MockAuthConnector implements IAuthConnector {
+    public authenticate(token: string): Promise<any> {
+        return new Promise((resolve, reject) => {
+            resolve({
+                CanImpersonate: false,
```
Comment (s-henney) [inline src/graph/auth/auth.spec.ts]: Could you use one of our test users here? This guy's info is the same in all environments and won't change after a refresh:
CanImpersonate: false,
ContactId: 7777294,
DonorId: 7745938,
Email: "mpcrds+auto+2@gmail.com",
HouseholdId: 5819396,
ParticipantId: 7654359,
UserId: 4488274

File: src/graph/sites/sites.spec.ts [hunk 2]
```diff
@@ -0,0 +1,98 @@
...
+import "reflect-metadata";
+import { Types } from '../../ioc/types';
+import { MockAuthConnector } from '../auth/auth.spec';
+import { IAuthConnector } from '../auth/auth.interface';
+
+@injectable()
+export class MocksSitesConnector implements ISitesConnector {
+    public getSites(): Promise<ISite[]> {
+        return new Promise((resolve, reject) => {
+            resolve([
```
Comment (s-henney) [inline src/graph/sites/sites.spec.ts]: This array is the same as the spec except for id values being ints. Could you create a constant array you use in both class and spec, or does the int vs. string matter?

File: src/graph/sites/sites.spec.ts [hunk 3]
```diff
@@ -0,0 +1,98 @@
...
+
+    const server = new ApolloServer({
+        typeDefs: schema,
+        resolvers: resolvers,
+        context: () => (authConnector.authenticate('fakeTokenDoesntMatter')),
+    });
+
+    const { query } = createTestClient(server);
+    const res = await query({
+        query: `{
```
Comment (s-henney) [inline src/graph/sites/sites.spec.ts]: Does the query() param here just describe how the results should be mapped?
If so, what happens if the param contains a property that doesn't exist in the results? Will the results contain that property set to undefined/null/empty string? Or will the whole query fail?

File: src/graph/sites/sites.spec.ts (hunk 2 above)
Comment (janssejl) [reply src/graph/sites/sites.spec.ts]: it actually does matter for graphql oddly enough. They have their own ID type and it is always a string :(

File: src/graph/sites/sites.spec.ts (hunk 3 above)
Comment (janssejl) [reply src/graph/sites/sites.spec.ts]: Yes! This is the query the client will use to fetch data from this graphQL API. The whole query will fail. I added a test for this below.

Comment (janssejl) [reply src/graph/sites/sites.spec.ts]: So I guess to further expand, "query" not only describes how it should be mapped, it is the actual request for data. Think of it as the query params for a RESTful GET.

File: src/graph/sites/sites.spec.ts (hunk 2 above)
Comment (s-henney) [reply src/graph/sites/sites.spec.ts]: Fair enough, no change then.

File: src/graph/auth/auth.spec.ts (hunk 1 above)
Comment (janssejl) [reply src/graph/auth/auth.spec.ts]: Fixed. Thanks for the data!

**Pull request merged**